### ISSUE TITLE
Default enrollment profile & DEP profile should pin certs for checkin and connect

### ISSUE THREAD

**Issue by knightsc** [micromdm/micromdm#488]
The default profile already installs the MicroMDM root certificate. We should additionally set the installed root certificate as pinned for the ServerURL and CheckInURL specified in the profile. Below is the documentation on these two settings

**ServerURLPinningCertificateUUIDs**
*Optional*. Array of strings containing the PayloadUUIDs of
certificates to be used when evaluating trust to the
.../connect/ URLs of MDM servers.
*Availability*: Available in macOS 10.13 and later.

**CheckInURLPinningCertificateUUIDs**
*Optional*. Array of strings containing the PayloadUUIDs of
certificates to be used when evaluating trust to the
.../checkin/ URLs of MDM servers.
*Availability*: Available in macOS 10.13 and later.

Even though it's only available in 10.13 and later I would hope old clients simply ignore these extra parameters. This change should be tested on macOS versions older than 10.13 to make sure this is true though

**Comment by jessepeterson** [comment]
By root certificate you mean the web certificate? We might have an issue if folks are using Let's Encrypt — the certificates will be rotated out from under them. I suppose we could re-enroll clients to get ahead of the rotation — which will be often for LE or any other short-term certificate situtation.

I suppose one question is: will these pinning keys do certificate-_only_ pinning, or can we also pin CAs? If the latter then this gets a little more flexible.

As well we might consider by default using the `anchor_certs` parameter in the DEP profile. We should verify, perhaps with Apple, that this truly does the the docs say: either the system roots _OR_ use the `anchor_certs` in evaluating trust. The latter would function as, effectively, CA pinning.

**Comment by knightsc** [comment]
Yeah I think I was thinking of the CA certificate which is installed. In general I wanted to at least get something submitted to look into all of these and see how it works. I suppose it is already something users could set on their own by editing their enrollment config and re-uploading. I was just trying to think if there was some way out of the box we could support those settings.
